### PR TITLE
feat(auth): add effects

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/effects/auth.effects.spec.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    email = mockRegistration.customer.email;
+    password = mockRegistration.password;
+    mockLoginInfo = {email, password};
+  });
+
+  it('should be created', () => {
+    expect(effects).toBeTruthy();
+  });
+
+  describe('when AuthLoginAction is triggered', () => {
```
Comment (lderrickable) [inline libs/auth/src/effects/auth.effects.spec.ts:98]: I think we should try to write the describe part of the tests so it's a little easier to discern the purpose of the code. For example, here:
```
describe - login$ - when the user logs in.
   describe - and the login is successful.
       it - should notify state that the login was successful.
   describe - and the login fails.
       it - should notify state that the login failed.
```

Comment (lderrickable) [reply libs/auth/src/effects/auth.effects.spec.ts:98]: lgtm otherwise

Comment (griest024) [reply libs/auth/src/effects/auth.effects.spec.ts]: changed

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**